phase4-lib/src/main/java/com/helger/phase4/sender/AbstractAS4PullRequestBuilder.java

```diff
@@ -98,6 +116,62 @@ public final IMPLTYPE userMsgConsumer (@Nullable final IAS4UserMessageConsumer a
     return thisAsT ();
   }
 
+  /**
+   * @return The currently set P-Mode. May be <code>null</code>.
+   */
+  @Nullable
+  public final IPMode pmode ()
+  {
+    return m_aPMode;
+  }
+
+  /**
+   * Set the PMode to be used. By default a generic PMode is used.
+   *
+   * @param aPMode
+   *        The PMode to be used. May be <code>null</code>.
+   * @return this for chaining
+   */
+  @Nonnull
+  public final IMPLTYPE pmode (@Nullable final IPMode aPMode)
+  {
+    if (aPMode == null)
+      LOGGER.warn ("A null PMode was supplied");
+    m_aPMode = aPMode;
+    return thisAsT ();
+  }
+
+  /**
+   * Set an optional Ebms3 Signal Message Consumer. If this consumer is set, the
+   * response is trying to be parsed as a Signal Message. This method is
+   * optional and must not be called prior to sending.
+   *
+   * @param aSignalMsgConsumer
+   *        The optional signal message consumer. May be <code>null</code>.
+   * @return this for chaining
+   */
+  @Nonnull
+  public final IMPLTYPE signalMsgConsumer (@Nullable final IAS4SignalMessageConsumer aSignalMsgConsumer)
+  {
+    m_aSignalMsgConsumer = aSignalMsgConsumer;
+    return thisAsT ();
+  }
+
+  /**
+   * Set an optional Attachment Consumer. This method is
+   * optional and must not be called prior to sending.
+   *
+   * @param aAttachmentConsumer
+   *        The optional attachment consumer. May be <code>null</code>.
+   * @return this for chaining
+   */
+  @Nonnull
+  public final IMPLTYPE attachmentConsumer (@Nullable final IAS4AttachmentConsumer aAttachmentConsumer)
```

Comment:
Same question as for userMessages - as the attachments are already in the `IAS4MessageState` why register a separate handler for that?